frontend/src/components/AppMap.vue

```diff
@@ -1,42 +1,109 @@
 <template>
-  <div
-    ref="element"
-    class="map"
-    :style="{
-      width: width ? width + 'px' : '100%',
-      height: height ? height + 'px' : '100%',
-    }"
-  />
-
-  <Teleport v-if="legendTeleport && showLegend" :to="legendTeleport">
-    <div class="legend" @mousedown.stop>
-      <div class="header">
-        <slot name="legend" />
+  <div class="container">
+    <div ref="scroll" class="scroll">
+      <!-- map container -->
+      <div
+        ref="element"
+        class="map"
+        :style="{
+          width: width ? width + 'px' : '100%',
+          height: height ? height + 'px' : '100%',
+        }"
+      />
+    </div>
+
+    <!-- control set -->
+    <div class="controls">
+      <AppButton
+        v-tooltip="'Download current map view as PNG'"
+        :icon="faDownload"
+        :accent="true"
+        @click="download"
+        >Download Map</AppButton
+      >
+      <AppButton
+        v-tooltip="'Zoom out'"
+        :icon="faMinus"
+        @click="map?.zoomOut()"
+      />
+      <AppButton v-tooltip="'Zoom in'" :icon="faPlus" @click="map?.zoomIn()" />
+      <AppButton
+        v-tooltip="'Fit view to data'"
+        :icon="faCropSimple"
+        @click="fit"
+        >Fit</AppButton
+      >
+      <AppButton
```

Comment:
Moved controls to within component so don't have to expose funcs and such to parent.

Added fullscreen button very easily thanks to vue-use.